### PR TITLE
Adds more device data to a user review (#1137) [/2]

### PR DESCRIPTION
**Pull request by andywerner**

### CODE REVIEW

File: octoprint_mrbeam/static/js/review.js [hunk 1]
```diff
@@ -147,16 +153,19 @@ $(function () {
         self.sendReviewToServer = function () {
             let review = $("#review_textarea").val();
             let data = {
+                // more data is added by the backend
                 dontShowAgain: self.dontShowAgain(),
                 rating: self.rating(),
                 review: review,
                 ts: new Date().getTime(),
-                env: MRBEAM_ENV_LOCAL,
                 sw_tier: MRBEAM_SW_TIER,
```
Comment (amrsoll) [inline octoprint_mrbeam/static/js/review.js]: Not necessary anymore

File: octoprint_mrbeam/static/js/review.js [hunk 2]
```diff
@@ -147,16 +153,19 @@ $(function () {
...
-                snr: MRBEAM_SERIAL,
                 number: self.REVIEW_NUMBER,
             };
 
+            if (FORCE_REVIEW_SCREEN) {
+                data["debug"] = true;
+            }
```
Comment (amrsoll) [inline octoprint_mrbeam/static/js/review.js:166]: Why not this :
```js
data ={
  ...
  debug: FORCE_REVIEW_NUMBER
)
```

File: octoprint_mrbeam/analytics/review_handler.py [hunk 3]
```diff
@@ -48,11 +49,28 @@ def _on_mrbeam_plugin_initialized(self, event, payload):
...
+        given = self._settings.get(["review", "given"])
+        if not given or data.get("debug", False):
+            data = self._add_review_data(data)
+            self._write_review_to_file(data)
+            self._settings.set_boolean(["review", "given"], data["dontShowAgain"])
+            self._settings.save()  # This is necessary because without it the value is not saved
+
+            ReviewFileUploader.upload_now(self._plugin, self._review_lock)
+        else:
+            self._logger.warn("Not accepting user review since it was already given.")
```
Comment (amrsoll) [inline octoprint_mrbeam/analytics/review_handler.py:61]: How about updating the dataset with `data["already_given"] = given` and filter afterwards, perhaps to see if the review changed

**Review by amrsoll — APPROVED** [review]

File: octoprint_mrbeam/static/js/review.js (hunk 1 above)
Comment (andywerner) [reply octoprint_mrbeam/static/js/review.js]: true

File: octoprint_mrbeam/analytics/review_handler.py (hunk 3 above)
Comment (andywerner) [reply octoprint_mrbeam/analytics/review_handler.py:61]: Each device should only send one review ever (only exception: if the device got reset). So far this was only enforced on the frontend side while the backend accepted happily as many reviews as the frontend (or a dev tools hacker) submitted. This is just to enforce this policy on the backend as well. Therefor I thinks it's ok this way.

File: octoprint_mrbeam/static/js/review.js (hunk 2 above)
Comment (andywerner) [reply octoprint_mrbeam/static/js/review.js:166]: se my comment above

**Pull request merged**